app/build.gradle

```diff
@@ -41,62 +40,61 @@ android {
 }
 
 ext {
-    supportLibVersion = '27.1.1'
-    exoPlayerLibVersion = '2.8.2'
+    supportLibVersion = '28.0.0'
+    exoPlayerLibVersion = '2.8.4' //2.9.0
     roomDbLibVersion = '1.1.1'
-    leakCanaryLibVersion = '1.5.4'
-    okHttpLibVersion = '3.10.0'
+    leakCanaryLibVersion = '1.5.4' //1.6.1
+    okHttpLibVersion = '3.11.0'
     icepickLibVersion = '3.2.0'
     stethoLibVersion = '1.5.0'
 }
 
 dependencies {
-    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2') {
+    androidTestImplementation('com.android.support.test.espresso:espresso-core:3.0.2', {
         exclude module: 'support-annotations'
-    }
+    })
 
-    implementation 'com.github.TeamNewPipe:NewPipeExtractor:66c3c3f45241d4b0c909'
+    implementation 'com.github.TeamNewPipe:NewPipeExtractor:24cb684e2353'
 
     testImplementation 'junit:junit:4.12'
-    testImplementation 'org.mockito:mockito-core:2.8.9'
+    testImplementation 'org.mockito:mockito-core:2.23.0'
 
-    implementation "com.android.support:appcompat-v7:$supportLibVersion"
-    implementation "com.android.support:support-v4:$supportLibVersion"
-    implementation "com.android.support:design:$supportLibVersion"
-    implementation "com.android.support:recyclerview-v7:$supportLibVersion"
-    implementation "com.android.support:preference-v14:$supportLibVersion"
+    implementation "com.android.support:appcompat-v7:${supportLibVersion}"
+    implementation "com.android.support:support-v4:${supportLibVersion}"
+    implementation "com.android.support:design:${supportLibVersion}"
+    implementation "com.android.support:recyclerview-v7:${supportLibVersion}"
+    implementation "com.android.support:preference-v14:${supportLibVersion}"
+    implementation "com.android.support:cardview-v7:${supportLibVersion}"
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
 
-    implementation 'ch.acra:acra:4.9.2'
+    implementation 'ch.acra:acra:4.9.2' //4.11
```

Comment:
Why did we add newer versions as comments here and somewhere below?

Reply:
because they didn't seem to work with ci